src/utils/Cypher.ts

```diff
@@ -161,4 +174,82 @@ export default class Cypher {
     // Save
     await SettingStorage.saveSettings(tenantID, settingsToSave);
   }
+
+  private static async getCryptoSettings(tenantID: string): Promise<CryptoSettings> {
+    const cryptoSettings = await SettingStorage.getCryptoSettings(tenantID);
+    if (!cryptoSettings || !cryptoSettings.crypto) {
+      throw new BackendError({
+        source: Constants.CENTRAL_SERVER,
+        module: MODULE_NAME,
+        method: 'getCryptoSetting',
+        message: `Tenant ID '${tenantID}' does not have crypto settings.`
+      });
+    }
+    return cryptoSettings;
+  }
+
+  private static async getSettingsWithSensitiveData(tenantID: string): Promise<SettingDB[]> {
+    // Get all settings per tenant
+    const settings = await SettingStorage.getSettings(tenantID, {},
+      Constants.DB_PARAMS_MAX_LIMIT);
+    // Filter settings with sensitiveData
+    return settings.result.filter((value: SettingDB) => {
+      if (value?.sensitiveData && !Utils.isEmptyArray(value?.sensitiveData)) {
+        return true;
+      }
+    });
+  }
+
+  private static async migrateSettings(tenantID: string, cryptoSetting: CryptoSetting): Promise<void> {
+    const settingsToMigrate = await Cypher.getSettingsWithSensitiveData(tenantID);
+    // If tenant has settings with sensitive data, migrate them
+    if (!Utils.isEmptyArray(settingsToMigrate)) {
+      // Migrate
+      for (const settingToMigrate of settingsToMigrate) {
+        // Migration already done?
+        if (Utils.isEmptyArray(settingToMigrate.backupSensitiveData)) {
+          // Save former senitive data in setting
+          const backupSensitiveData = Cypher.prepareBackupSensitiveData(settingToMigrate);
+          backupSensitiveData.push(Cypher.hash(cryptoSetting.formerKey));
```

Comment:
No need to store the key and it is already stored in crypto settings.
Just removed it.

Reply:
Sorry I removed it, but then forgot to pull and when merging I lost it :(